### PR TITLE
Swap deprecated getchildren() for list()

### LINKED ISSUE #56475: deprecation warning for getchildren when running salt-cloud
### Description of Issue

When I use `salt-cloud` to create or destroy a minion (in EC2), I get the following three warnings from the same line of code:

```
[WARNING ] /usr/lib/python3/dist-packages/salt/utils/xmlutil.py:30: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  if len(xmltree.getchildren()) < 1:

[WARNING ] /usr/lib/python3/dist-packages/salt/utils/xmlutil.py:30: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  if len(xmltree.getchildren()) < 1:

[WARNING ] /usr/lib/python3/dist-packages/salt/utils/xmlutil.py:30: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  if len(xmltree.getchildren()) < 1:
```

This is not fatal, of course. The minion is created or destroyed correctly. 

A note about the deprecation can be seen in the docs [here](https://docs.python.org/3.7/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.getchildren).

A search through the develop branch shows additional use of `getchildren` in `salt/modules/bamboohr.py`, `salt/modules/win_lgpo.py`, and `salt/utils/aws.py`. I haven't yet encountered the warning from these files. (The master branch also has it in `salt/cloud/clouds/ec2.py`.)

I can suppress the warnings by changing e.g. `xmltree.getchildren()` to `list(xmltree)` in `xmlutil.py`. I'd be happy to submit a PR to remove uses of `getchildren` in this and the other files, but I thought I'd file a ticket first in case there are aspects I haven't thought of.

### Setup
Working cloud providers and profiles for AWS.

### Steps to Reproduce Issue

Run
```
sudo salt-cloud --out=json --script-args="-xpython3" -p <profile> <minion_id>
```

### Versions Report

```
Salt Version:
           Salt: 3000
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: unknown
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.11
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.7.3 (default, Dec 20 2019, 18:57:59)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 17.1.2
          smmap: 2.0.5
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist: debian 10.3 
         locale: UTF-8
        machine: x86_64
        release: 4.19.0-8-amd64
         system: Linux
        version: debian 10.3 
```

### PR DESCRIPTION
**Pull request by bensteinberg**
I've closed #56482 and opened this PR in order to rebase the changes on master.

### What does this PR do?

This PR replaces the deprecated `getchildren()` with `list()`. See a note about the deprecation [here](https://docs.python.org/3.7/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.getchildren).
### What issues does this PR fix or reference?

Closes #56475.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by waynew** [comment]
@dwoz in theory this should be covered by existing tests. In practice, I suspect there are a lack of tests.

@bensteinberg Looks like there's some merge conflicts going on here.

**Comment by DmitryKuzmenko** [comment]
To be sure we have to check the coverage results of this. Since there no any logic change the only urgent point here is to ensure tests cover the code (each line ran at least once). Just to be sure there're no mistypes and so on.

**Comment by DmitryKuzmenko** [comment]
@bensteinberg could you please resolve conflicts on this?

**Comment by bensteinberg** [comment]
Will do, stay tuned.

**Comment by bensteinberg** [comment]
Sorry for the delay here.

File: salt/modules/win_lgpo.py [hunk 1]
```diff
@@ -5738,6 +5738,7 @@ def _checkListItem(
     for list_element in xpath_object(policy_element):
         configured_items = 0
         required_items = 0
+
```
Comment (DmitryKuzmenko) [inline salt/modules/win_lgpo.py:5741]: You've changed nothing in this file. Could you please revert these empty lines to not include it to the PR?

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by DmitryKuzmenko** [comment]
There are some tests failures related to pycrypto that I've fixed in #57519 
This should be good after #57519 is merged.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
This needs testcase

**Comment by bensteinberg** [comment]
@DmitryKuzmenko, thanks for the fix to `win_lgpo.py` and #57519.

**Comment by cmcmarrow** [comment]
If you need help with test writing please message me on slack `saltstackcommunity.slack.com` my user name is `Chad McMarrow`. I will be happy to have a zoom meeting with you and show you how to write and setup tests.

**Comment by cmcmarrow** [comment]
Hi, @bensteinberg. Thanks for the PR!. I will re-run all the tests once #57519 is merged.

**Comment by sagetherage** [comment]
looks like this was failing on pre-commit. Here is what we recommend:

For setting up pre-commit, check the following guidance:

https://docs.saltproject.io/en/master/topics/development/contributing.html#pre-commit-and-nox-setup
This depends your dev environment setup, and this link shows you how to install pre-commit via pip, and then run pre-commit install in the root of your local salt repo forky.

Once you've followed those directions, you need to run the following to apply any fixes or guidance against the files you are modifying:

pre-commit run --files salt/modules/rh_ip.py salt/states/network.py
After this command runs and applies changes, now create a new git commit that is taking all of the newly applied changes.

In the future, pre-commit will handle changes automatically without you having to run this command because it will run before any git commit is able to complete. :)

If you run into any more problems around pre-commit, feel free to ask. Additional assistance can be found on our Slack channel, if needed, also:

https://saltstackcommunity.herokuapp.com/

File: salt/cloud/clouds/ec2.py [hunk 2]
```diff
@@ -216,7 +216,7 @@ def _xml_to_dict(xmltree):
             name = comps[1]
         if name not in xmldict:
             if sys.version_info < (2, 7):
-                children_len = len(item.getchildren())
+                children_len = len(list(item))
```
Comment (s0undt3ch) [inline salt/cloud/clouds/ec2.py:219]: This whole `if` block should be removed. Salt doesn't support Py2 anymore, let alone 2.6.

Just use the `else` clause, `children_len = len(item)`

File: salt/cloud/clouds/ec2.py [hunk 3]
```diff
@@ -466,7 +466,7 @@ def query(
 
     if setname:
         if sys.version_info < (2, 7):
-            children_len = len(root.getchildren())
+            children_len = len(list(root))
```
Comment (s0undt3ch) [inline salt/cloud/clouds/ec2.py:469]: Same as above

File: salt/utils/aws.py [hunk 4]
```diff
@@ -548,7 +548,7 @@ def query(
 
     if setname:
         if sys.version_info < (2, 7):
-            children_len = len(root.getchildren())
+            children_len = len(list(root))
```
Comment (s0undt3ch) [inline salt/utils/aws.py:551]: Same as the comments on the ec2 file

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
Basically, just remove the Py2 code from both modules and we're good.

**Comment by bensteinberg** [comment]
Thanks, @sagetherage and @s0undt3ch -- I think I have my environment/pre-commit set up right, and have just force-pushed a new commit. I hope that's okay.

**Comment by s0undt3ch** [comment]
> I hope that's okay.

Yep, that's the way, clean commits.

Thank You!

File: salt/cloud/clouds/ec2.py [hunk 5]
```diff
@@ -465,12 +454,7 @@ def query(
         items = root
 
     if setname:
-        if sys.version_info < (2, 7):
-            children_len = len(root.getchildren())
-        else:
-            children_len = len(root)
-
-        for item in range(0, children_len):
+        for item in enumerate(root):
```
Comment (s0undt3ch) [inline salt/cloud/clouds/ec2.py:457]: The code you have will fail with a traceback or mess with the logic.

Comment (s0undt3ch) [reply salt/cloud/clouds/ec2.py:457]:
```python
for idx, item in enumerate(root):
    comps = item.tag.split("}")
    if comps[1] == setname:
        items = root[idx]
```

File: salt/utils/aws.py [hunk 6]
```diff
@@ -547,12 +538,7 @@ def query(
         items = root
 
     if setname:
-        if sys.version_info < (2, 7):
-            children_len = len(root.getchildren())
-        else:
-            children_len = len(root)
-
-        for item in range(0, children_len):
+        for item in range(0, len(root)):
```
Comment (s0undt3ch) [inline salt/utils/aws.py:541]: Apply the same enumerate logic I commented above.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/aws.py (hunk 6 above)
Comment (bensteinberg) [reply salt/utils/aws.py:541]: Thanks, @s0undt3ch -- I will fix this and the other one. I won't be able to get to this for a few days.

Comment (bensteinberg) [reply salt/utils/aws.py]: @s0undt3ch, I've just rebased, fixed according to your notes (I think), and force-pushed. Thanks for your patience.

File: salt/cloud/clouds/ec2.py [hunk 7]
```diff
@@ -220,13 +214,8 @@ def _xml_to_dict(xmltree):
         if "}" in name:
             comps = name.split("}")
             name = comps[1]
-        if name not in xmldict:
```
Comment (s0undt3ch) [inline salt/cloud/clouds/ec2.py:223]:
```suggestion
        if name not in xmldict:
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
One last change

File: salt/cloud/clouds/ec2.py (hunk 7 above)
Comment (s0undt3ch) [reply salt/cloud/clouds/ec2.py:223]: This line can't be deleted

Comment (bensteinberg) [reply salt/cloud/clouds/ec2.py:223]: Wow, I'm not sure how I managed that. I've just restored the line.

**Review by s0undt3ch — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by krionbsd — APPROVED** [review]

**Pull request merged**